woocommerce/pinterest-for-woocommerce
# Add/switch feed v5

### Changes proposed in this Pull Request:

Closes #833 .

Adds Pinterest API v5 integration into feeds.
Merging `develop` in as well to catch up on fresh phpcs and unit tests setups.

### Detailed test instructions:

1. Checkout `add/switch-feed-v5` branch.
2. Check Product Sync being at Settings tab.

<img width="1293" alt="Settings_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/74e8ae87-12fd-4a91-b208-df0036cf6365">

3. Go to _WooCommerce - Status - Scheduled Actions - Pending_ tab and force run `pinterest-for-woocommerce-handle-feed-registration` action.

<img width="1293" alt="WooCommerce_status_‹_WordPress_Pinterest_—_WordPress" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/7c807267-b0c3-4659-bd56-2cded04df1b8">

4. Go to Pinterest and check for the feed to be present and active under _Catalogues - Data Sources_

- [ ] Feed name must be like _"Created by Pinterest for WooCommerce [country(Site Language)]-[currency]."_
- [ ] Feed must have _Active_ Status

**NOTE:** Checks feed's Language column, it must be the same you have under your WordPress Settings.

<img width="1293" alt="Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/39eed1c6-e78a-44bc-b341-74d81b007be3">

<img width="1293" alt="Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/d9d5fe76-d7c3-4c2d-8cb9-ca9fd000469c">

5. Go to WordPress Settings and change **_Site Language_** to something else of your choice. I switched between _US_ and _UK_ English languages.

<img width="1293" alt="General_Settings_‹_WordPress_Pinterest_—_WordPress" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/12d83aef-97fa-4edd-aeeb-a53857323400">

6. Check Pinterest _Catalogues - Data Sources_ for another feed to be added and activated.

<img width="1293" alt="Pinterest" src="https://github.com/woocommerce/pinterest-for-woocommerce/assets/9010963/f5a5d9e8-0c39-4e46-b611-41a0ef50b1c2">

### Changelog entry

> Add - Pinterest API v5 feeds integration.
